src/game/boe.actions.cpp

```diff
@@ -495,18 +485,9 @@ void handle_pause(bool& did_something, bool& need_redraw) {
 void handle_look(location destination, bool right_button, eKeyMod mods, bool& need_redraw, bool& need_reprint) {
 	if(recording){
 		std::map<std::string,std::string> info;
-		std::ostringstream sstr;
-		sstr << destination;
-		info["destination"] = sstr.str();
-
-		sstr.str("");
-		sstr << std::boolalpha << right_button;
-		info["right_button"] = sstr.str();
-
-		sstr.str("");
-		sstr << mods;
-		info["mods"] = sstr.str();
-
+		info["destination"] = boost::lexical_cast<std::string>(destination);
+		info["right_button"] = boost::lexical_cast<std::string>(right_button);
```

Comment:
Okay, but doesn't this return to representing true/false as 1/0?

Reply:
I guess it does. After some searching, I don't see very good options for getting lexical_cast to do "true"/"false", except maybe this: https://stackoverflow.com/a/17708833

Reply:
That was the same answer I found… the problem with it is that we'd have to make a policy to never include lexical_cast.hpp directly.

We could either:

1. Do like the scenario format – save using `b ? "true" : "false"` and load as `str == "true"` (meaning any other invalid value ends up as false).
2. Create a custom function that's basically lexical_cast but deals in true/false instead of 1/0. Probably use it in the scenario format too.

Reply:
It'd be trivial to make 2 global inline functions out of the snippets from your option 1. I think that's what we should do--it seems kind of absurd to do anything more complicated than that.

Reply:
Although now I can't think of a surefire way to find all the instances of lexical_cast<string>(bool) conversions and change them. for string->bool I can grep to find every instance, but not the other way around.

Reply:
Did you manage to find them all? If it's so easy to find the casts _to_ bool then I think it should probably suffice to find the opposite cast for each of those…

Reply:
I just went through every boost::lexical_cast<std::string> and manually identified bool cases.